storage/innobase/include/fil0crypt.h

```diff
@@ -612,6 +612,41 @@ fil_space_get_scrub_status(
 	const fil_space_t*		space,
 	fil_space_scrub_status_t*	status);
 
+/** Handles the fetching/generation/storing/etc of keyring redo log keys.
+ 
+ This class is *NOT* thread safe, as thread safety is not required.
+ Data is only accessed/modified on the following points:
+ * When the redo space is created, at startup
+ * During redo log recovery, at startup
+ * When the server UUID is generated, at startup
+ * When the user requests a new key version, in the setter 
```

Comment:
Why it is OK to be thread unsafe, if two users can request new key versions in parallel?

Reply:
New key versions are added only in the master thread, which checks it periodically.

Reply:
Done